src/Symfony/Bundle/FrameworkBundle/Controller/ControllerNameParser.php

```diff
@@ -47,7 +47,12 @@ public function __construct(KernelInterface $kernel)
     public function parse($controller)
     {
         $originalController = $controller;
-        if (3 !== count($parts = explode(':', $controller))) {
+        $parts = explode(':', $controller);
+
+        // The second condition is for cases when this method is called with the
```

Comment:
I think the exception message is already straightforward now? Imo. no need for this comment.